subsys/mpsl/cx/thread/mpsl_cx_thread.c

```diff
@@ -112,14 +110,12 @@ static void gpiote_irq_handler(const struct device *gpiob, struct gpio_callback
 	}
 }
 
-static int32_t gpio_init(uint8_t pin_no, bool input, const struct device **port,
-		gpio_port_value_t *pin_mask)
+static int32_t gpio_init(uint8_t pin_no, bool input, const char *gpio_lbl,
```

Comment:
I'm thinking if how this function actually simplifies or complicates the code. I think it's main purpose was abstracting `use_port_1`-related logic.

Reply:
Still, the same sequence of operations needs to be done for all three pins of the interface. I think it's better to have it in a single function.